build.py

```diff
@@ -1653,14 +1715,34 @@ def __copy(self, src, destdir):
             print_debug("copying '%s' to '%s'" % (src, destdir))
             shutil.copy(src, destdir)
 
+    def __hashfile(self, file_name):
+        hash_calc = hashlib.sha256()
+        with open(file_name, 'rb') as fi:
+            for chunk in iter(lambda: fi.read(4096), b""):
+                hash_calc.update(chunk)
+        return hash_calc.hexdigest()
+
+    def __check_hash(self, proj):
+        if hasattr(proj, 'hash'):
+            hc = self.__hashfile(proj.archive_file)
+            if hc != proj.hash:
+                print_message("Hash mismatch on %s:\n  Calculated '%s'\n  Expected   '%s'\n" % (proj.archive_file, hc, proj.hash, ))
+                return 1
+            else:
+                if self.opts.check_hash:
+                    print_message("Hash ok on %s (%s)" % (proj.archive_file, hc, ))
+                else:
+                    print_debug("Hash ok on %s (%s)" % (proj.archive_file, hc, ))
+        return 0
+
     def __download_one(self, proj):
         if not proj.archive_file:
             print_debug("archive file is not specified for project %s, skipping" % (proj.name,))
-            return
+            return 0

```

Comment:
For the last changes I delete & redo the pull request. Now I only make the change, pull on my repository and make/update a pull request: there is no confirmation but it seem that the change is now visible to you. Is correct or I have to do something more ?

Thanks.